forui/lib/src/widgets/pagination/pagination_style.dart

```diff
@@ -0,0 +1,80 @@
+import 'package:flutter/foundation.dart';
+import 'package:flutter/widgets.dart';
+
+import 'package:meta/meta.dart';
+
+import 'package:forui/forui.dart';
+
+part 'pagination_style.style.dart';
+
+/// A [FPagination] state's style.
+final class FPaginationStateStyle with Diagnosticable, _$FPaginationStateStyleFunctions {
+  /// The decoration applied to the pagination item.
+  @override
+  final BoxDecoration decoration;
+
+  /// The decoration applied when the pagination item is hovered.
+  @override
+  final BoxDecoration hoveredDecoration;
+
+  /// The text style.
+  @override
+  final TextStyle textStyle;
+
+  /// Creates a [FPaginationStateStyle].
+  FPaginationStateStyle({required this.decoration, required this.hoveredDecoration, required this.textStyle});
+}
+
+/// The [FPagination] styles.
+final class FPaginationStyle with Diagnosticable, _$FPaginationStyleFunctions {
+  /// The style for the [FPagination] when it is selected.
+  @override
+  final FPaginationStateStyle selected;
+
+  /// The style for the [FPagination] when it is unselected.
+  @override
+  final FPaginationStateStyle unselected;
+
+  /// The icon style.
+  @override
+  final FIconStyle iconStyle;
+
+  /// The padding around each item. Defaults to EdgeInsets.symmetric(horizontal: 2)`.
+  @override
+  final EdgeInsets itemPadding;
+
+  /// The constraints for the content. Defaults to `BoxConstraints(maxWidth: 40, minWidth: 40, maxHeight: 40, minHeight: 40)`.
+  @override
+  final BoxConstraints contentConstraints;
+
+  /// Creates a [FPaginationStyle].
+  FPaginationStyle({
+    required this.selected,
+    required this.unselected,
+    required this.iconStyle,
+
+    this.itemPadding = const EdgeInsets.symmetric(horizontal: 2),
+    this.contentConstraints = const BoxConstraints(maxWidth: 40.0, minWidth: 40.0, maxHeight: 40.0, minHeight: 40.0),
```

Comment:
`https://api.flutter.dev/flutter/rendering/BoxConstraints/BoxConstraints.tightFor.html` can be used instead. (please update docs too).